src/librustc_metadata/cstore_impl.rs

```diff
@@ -277,7 +277,9 @@ pub fn provide_local<'tcx>(providers: &mut Providers<'tcx>) {
             Rc::new(link_args::collect(tcx))
         },
         extern_mod_stmt_cnum: |tcx, id| {
-            let id = tcx.hir.definitions().find_node_for_hir_id(id);
+            assert_eq!(id.krate, LOCAL_CRATE);
+            let hir_id = tcx.hir.definitions().def_index_to_hir_id(id.index);
+            let id = tcx.hir.definitions().find_node_for_hir_id(hir_id);
```

Comment:
Whoops, I should better have listened to @arielb1 and marked `find_node_for_hir_id` as super-slow somehow. It does a linear search over all NodeIds :cold_sweat:

But it seems that what you're doing here is equivalent to `tcx.hir.as_local_node_id(id).unwrap()`?

Reply:
Why does `find_node_for_hir_id` exist? I've introduced `hir_to_node_id`.

Reply:
Oh, I see. It does exist because I wanted to avoid building that reverse mapping because it was only needed in error reporting. It's probably cheap to build though, especially now that locals don't have DefIds anymore.